cmd/deploy_test.go

```diff
@@ -903,3 +903,201 @@ func TestAPIModelWithContainerMonitoringAddonWithWorkspaceGuidAndKeyConfigInCmd(
 		t.Fatalf("expected workspaceKey : %s but got : %s", expectedWorkspaceKeyInBase64, addon.Config["workspaceKey"])
 	}
 }
+
+func TestAPIModelWithContainerMonitoringAddonWithWorkspaceGuidAndKeyConfigInCmdInMooncake(t *testing.T) {
```

Comment:
I love the test coverage, but I must ask you to dry up this code a bit. Please introduce these tests as a single test with cases. A good example of this is: https://github.com/Azure/aks-engine/blob/bdd4cf5622f0b31a16be60ce48d8a6fb7499c31a/cmd/deploy_test.go#L161-L251.

Reply:
I notice you added cases to the other test, so you don't need an example :). Sorry if that was a little pedantic.

Reply:
Hi, @devigned, I refactored the tests to have single test with different cases. can you please look at the changes and also help running pr e2e tests on this?